From 371323cb8e0558f7577884a81ebcad4a33c3c442 Mon Sep 17 00:00:00 2001
From: "Lin, Pei-Hung" <lin32@llnl.gov>
Date: Fri, 27 Mar 2026 00:25:07 -0700
Subject: [PATCH] [MCAD Patch 0] Add identifier field to mca::Instruction

---
 llvm/include/llvm/MCA/Instruction.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/llvm/include/llvm/MCA/Instruction.h b/llvm/include/llvm/MCA/Instruction.h
index b6b5b5979dec..855e06335bd0 100644
--- a/llvm/include/llvm/MCA/Instruction.h
+++ b/llvm/include/llvm/MCA/Instruction.h
@@ -646,6 +646,9 @@ class Instruction : public InstructionBase {
   // True if this instruction has been optimized at register renaming stage.
   bool IsEliminated;
 
+  // Client-dependent identifier for this instruction.
+  std::optional<uint64_t> Identifier;
+
 public:
   Instruction(const InstrDesc &D, const unsigned Opcode)
       : InstructionBase(D, Opcode), Stage(IS_INVALID),
@@ -693,6 +696,9 @@ public:
   bool isRetired() const { return Stage == IS_RETIRED; }
   bool isEliminated() const { return IsEliminated; }
 
+  std::optional<uint64_t> getIdentifier() const { return Identifier; }
+  void setIdentifier(uint64_t Id) { Identifier = Id; }
+
   // Forces a transition from state IS_DISPATCHED to state IS_EXECUTED.
   LLVM_ABI void forceExecuted();
   void setEliminated() { IsEliminated = true; }
-- 
2.47.3

